db.go

```diff
@@ -174,3 +197,20 @@ func (self *DB) resp(method string, params []any, res any) (any, error) {
 	return res, nil
 
 }
+
+func isSlice(possibleSlice any) bool {
+
+	var x interface{} = possibleSlice
+
+	slice := false
+
+	switch v := x.(type) {
```

Comment:
Have a look into using reflection here, I think it may be more reliable.
In the case where a non-empty slice is passed through here, `("%s", x.(type))`, will not return true in any of the if checks you have below, and thus this function will result in a false, and code above will attempt to unmarshal in the non-slice state, which I don't think is expected..